File: Euler2.js
//Euler2
let a=[1]
let n=2
while(n<4000000){
	a.push(n)
	n=a[a.length-2]+a[a.length-1]
}
let r=a.map(x=>x%2==0?x:0).reduce((xs,x)=>xs+x)
Console.log(r)

File: Euler2.hs
fib n | n == 0 = 1
      | n == 1 = 1
      | otherwise = fib (n-1) + fib (n-2)

euler2 n = sum (takeWhile (< n) [fib x | x <- [0..], even (fib x)])

main = do
  print (euler2 4000000)
